tools/windows/edit-iso.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -efo pipefail
+
+usage() {
+    echo "Usage: $(basename "$0") [OPTIONS]
+
+This script modifies a Windows installation ISO image.
+
+Options:
+    --input   Input (unmodified) .iso file path
+    --output  Output file path
+    --files   Directory containing files to add (should contain autounattend.xml)
+              Installed partition's root directory corresponds to 'sources/\$OEM\$/\$1' on the image
+"
+}
+
+if ! OPTS=$(getopt -o hi:o:f: --long help,input:,output:,files: -n "$0" -- "$@"); then
+    exit 1
+fi
+
+eval set -- "$OPTS"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help) usage; exit 0 ;;
+        -i | --input) INPUT="$2"; shift ;;
+        -o | --output) OUTPUT="$2"; shift ;;
+        -f | --files) FILES="$2"; shift ;;
+    esac
+    shift
+done
+
+if [ -z "${INPUT}" ] || [ -z "${OUTPUT}" ] || [ -z "${FILES}" ]; then
+    usage
+    exit 1
+fi
+
+# qvm-run requires more RPC permissions
+# $1=target, $2=service, $3=input
+qrexec_call() {
+    readarray -t result <<< "$(echo -n "$3" | qrexec-client-vm "$1" "$2" | sed "s/\x00/\n/g")"
+    if [ "${result[0]}" != "0" ]; then
+        >&2 echo "qrexec call '$2' to '$1' failed: ${result[0]} ${result[1]}"
+        exit 1
+    fi
+    echo "${result[1]}"
+}
+
+# $1=target, $2=input
+shell_call() {
+    echo "$2" | qrexec-client-vm "$1" qubes.VMShell
+}
+
+SCRIPT_DIR=$(dirname "$0")
+SCRIPT_DIR=$(readlink -f "${SCRIPT_DIR}")
+
+echo "[*] Setting up a loop device for the ISO..."
+LODEV=$(losetup -f)
```

Comment:
and also this sometimes needs sudo (if there are no free devices and one needs to be created)

Reply:
Fixed.